cmd/api/main.go

```diff
@@ -49,9 +53,23 @@ func run() error {
 	//create and init http server:
 	server := bookhttp.NewServer(bookhttp.ServerConfig{Port: 8080}, bookHandler)
 
-	err = server.ListenAndServe()
-	if err != nil && !errors.Is(err, http.ErrServerClosed) {
-		return fmt.Errorf("unexpected http server error: %w", err)
+	go func() (err error) {
+		err = server.ListenAndServe()
+		if err != nil && !errors.Is(err, http.ErrServerClosed) {
+			return fmt.Errorf("unexpected http server error: %w", err)
+		}
+		return nil
+	}()
```

Comment:
I don't think this function should be returning an error. You never read it.
```suggestion

	go func() {
		err = server.ListenAndServe()
		if err != nil && !errors.Is(err, http.ErrServerClosed) {
			log.Println("unexpected http server error: %v", err)
		}
		log.Println("stopped serving new requests.")
	}()